src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1426,36 +1426,52 @@ void primitive_inst::do_runtime_skip_permute() {
     auto desc = _node->as<permute>().get_primitive();
     auto input_shape = _impl_params->get_input_layout(0).get_shape();
     const auto& permute_order = desc->permute_order;
-    // Check runtime shape
-    // Optimize when the largest value among the actual dim values in case where the permute order
-    // is different from the shape index is equal to the multiplied value
-    // Set size to zero to pass the case that permute order is same with input order like [0, 1, 2, 3]
-    int32_t size = 0;
-    int32_t max_value = 0;
-    for (int32_t i = 0; i < static_cast<int32_t>(permute_order.size()); ++i) {
-        int32_t order = static_cast<int32_t>(permute_order[i]);
-        int32_t dim = static_cast<int32_t>(input_shape[order]);
-        if (i != order) {
-            if (dim > max_value)
-                max_value = dim;
-            size = (size == 0) ? dim : (size * dim);
-        }
-    }
-
-    // If the largest value and total size are different, can_be_optimized needs to be reset
-    if (size != max_value) {
-        set_can_be_optimized(false);
-        GPU_DEBUG_TRACE_DETAIL << "--- Cannot optimize because size(" << size << ") and max_value(" << max_value
-                               << ") are different" << std::endl;
-
-        GPU_DEBUG_TRACE_DETAIL << "[do_runtime_skip_permute] " << id() << " : reset can_be_optimized to false "
-                               << std::endl;
-        return;
+    // Skippability
+    // 1. Focus on "congituous transpose range"
+    //    permute_order : [2, 1, 0]    => sub_ranges : permute_order[0, 1, 2]
+    //    permute_order : [1, 0, 2, 3] => sub_ranges : permute_order[0, 1, 2]
+    //    permute_order : [0, 2, 1, 3] => sub_ranges : permute_order[1, 2]
+    //    permute_order : [0, 3, 1, 2] => sub_ranges : permute_order[1, 2, 3]
+    //    permute_order : [2, 0, 1, 3] => sub_ranges : permute_order[0, 1, 2]
+    //    permute_order : [3, 2, 1, 0] => sub_ranges : permute_order[0, 1, 2, 3]
+    //    permute_order : [3, 2, 1, 0, 4] => sub_ranges : permute_order[0, 1, 2, 3]
+    //    permute_order : [0, 2, 1, 3, 5, 4] => sub_ranges : permute_order[1, 2], permute_order[4, 5]
+    //    permute_order : [4, 5, 2, 3, 0, 1] => sub_ranges : permute_order[0, 1, 2, 3, 4, 5]
+    // 2. Within each transpose range, only 1 non-1 dimension allowed at max.
+    //   i.e., for the following case,
+    //   permute_order : [0, 2, 1, 3, 5, 4] => sub_ranges : permute_order[1, 2], permute_order[4, 5]
+    //   if the input shape is [1, 1, 4, 3, 1, 4] => true because there is only one non-1 dim in both {input_shape[1], input_shape[2]}
+    //      and {input_shape[4], input_shape[5]}
+    //   if the input shape is [1, 2, 4, 3, 1, 4] => false because there is more than one non-1 dim in {input_shape[1], input_shape[2]}
+    size_t range_max_dim = 0;
```

Comment:
Is it supposed to optimize cases like permute_order = {1, 0, 3, 2} with input_shape = {20, 1, 30, 1}? It seems this case will not be optimized out because it is treated as a single sub-range instead of two different ones

Reply:
Good catch. I realized much more simpler method which covers all cases.